src/nvim/fileio.c

```diff
@@ -2310,6 +2310,10 @@ buf_write (
   char_u          *fenc_tofree = NULL;   /* allocated "fenc" */
 #ifdef HAS_BW_FLAGS
   int wb_flags = 0;
+#endif
+  char_u *fname_res = NULL; // De-symlinked filename
+#ifdef HAVE_READLINK
+  char_u fname_res_buf[MAXPATHL];
```

Comment:
could use `IObuff` instead of this new buffer.

Reply:
While looking up the definition of IObuff, I saw NameBuff right next to it and it seemed perfect for this situation:
`EXTERN char_u NameBuff[MAXPATHL];           ///< Buffer for expanding file names`

Can change if you prefer IObuff

Reply:
good thought, but `NameBuff` cannot be used in `buf_write`:

https://github.com/neovim/neovim/blob/2e171a6fd163e0d74c80d21a893cb6a3c3394b1f/src/nvim/fileio.c#L2241

Reply:
Darn globals...

Fixed.